scipy/optimize/minpack.py

```diff
@@ -449,23 +448,37 @@ def leastsq(func, x0, args=(), Dfun=None, full_output=0,
         return (retval[0], info)
 
 
-def _wrap_func(func, xdata, ydata, weights):
-    if weights is None:
+def _wrap_func(func, xdata, ydata, T):
```

Comment:
@dlax Thanks for going through this. @nmayorov came up with T for "transformation", basically depending upon whether you pass errors or covariance, T performs a transformation on the residuals to be optimized. Do you think adding some comments at the top of these functions will help?